cppcache/src/PdxType.cpp

```diff
@@ -568,16 +568,32 @@ bool PdxType::operator<(const PdxType& other) const {
   return false;
 }
 
-int32_t PdxType::hashcode() const {
+bool PdxType::operator==(const PdxType& other) const {
+  if (this->m_className != other.m_className){
+    return false;
+  }
+
+  if(this->m_noJavaClass != other.m_noJavaClass){
+    return false;
+  }
+
+  if(this->getTotalFields() != other.getTotalFields()){
+    return false;
+  }
+
+  //TODO: Compare m_pdxFieldTypes & other.m_pdxFieldTypes;
```

Comment:
What's with the new TODO?

Reply:
I wanted to separate in a different commit the check of the fields, as I was not sure how to do it. I was exploring the possibility to change `m_pdxFieldTypes` to be a set instead of a vector, for easy comparision of values, but that change had a lot of impacts. After that I realized I could use the `m_fieldNameVsPdxType` map for the comparision of the fields. Thinking about the order of the fields, better to compare two maps than two vectors. Next commit after this one removes this TODO